plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusClient.java

```diff
@@ -167,7 +169,15 @@ public byte[] fetchUri(URI uri)
         Request.Builder requestBuilder = new Request.Builder().url(uri.toString());
         try (Response response = httpClient.newCall(requestBuilder.build()).execute()) {
             if (response.isSuccessful() && response.body() != null) {
-                return response.body().bytes();
+                try (InputStream inputStream = response.body().byteStream()) {
```

Comment:
why would this work? At the end of the day don't we create a byte array of exact same size as before?

Reply:
Yep, this isn't an improvement at all